pkg/cri/opts/spec.go

```diff
@@ -51,27 +51,54 @@ func WithoutRoot(ctx context.Context, client oci.Client, c *containers.Container
 	return nil
 }
 
-// WithProcessArgs sets the process args on the spec based on the image and runtime config
-func WithProcessArgs(config *runtime.ContainerConfig, image *imagespec.ImageConfig) oci.SpecOpts {
-	return func(ctx context.Context, client oci.Client, c *containers.Container, s *runtimespec.Spec) (err error) {
-		command, args := config.GetCommand(), config.GetArgs()
-		// The following logic is migrated from https://github.com/moby/moby/blob/master/daemon/commit.go
-		// TODO(random-liu): Clearly define the commands overwrite behavior.
-		if len(command) == 0 {
-			// Copy array to avoid data race.
-			if len(args) == 0 {
-				args = append([]string{}, image.Cmd...)
+// getArgs is used to evaluate the overall args for the container by taking into account the image command and entrpoints
+// along with the container command and entrpoints specified through the podspec if any
+func getArgs(imgEntrypoint []string, imgCmd []string, ctrEntrypoint []string, ctrCmd []string) ([]string, bool, error) {
+	// firstArgFromImg is a flag that is returned to indicate that the first arg in the slice comes from either the image
+	// Entrypoint or Cmd. If the first arg instead comes from the container config (e.g. overriding the image values),
+	// it should be false.
+	// Essentially this means firstArgFromImg should be true iff:
+	// Ctr entrypoint	ctr cmd		image entrypoint	image cmd	firstArgFromImg
+	// --------------------------------------------------------------------------------
+	//	nil				 nil			exists			 nil		  true
+	//  nil				 nil		    nil				 exists		  true
+
+	// This is needed to support the non-OCI ArgsEscaped field used by Docker. ArgsEscaped is used for
+	// Windows images to indicate that the command has already been escaped and should be
+	// used directly as the command line.
+	var firstArgFromImg bool
+	entrypoint, cmd := ctrEntrypoint, ctrCmd
+	// The following logic is migrated from https://github.com/moby/moby/blob/master/daemon/commit.go
+	// TODO(random-liu): Clearly define the commands overwrite behavior.
+	if len(entrypoint) == 0 {
+		// Copy array to avoid data race.
+		if len(cmd) == 0 {
+			cmd = append([]string{}, imgCmd...)
+			if len(imgCmd) > 0 {
+				firstArgFromImg = true
 			}
-			if command == nil {
-				if !(len(image.Entrypoint) == 1 && image.Entrypoint[0] == "") {
-					command = append([]string{}, image.Entrypoint...)
-				}
+		}
+		if entrypoint == nil {
```

Comment:
This seems really odd to me. Why would a zero length be treated differently than nil here. IE: if len(entrypoint)== 0 and entrypoint != nil why would we not want the image entrypoint?